### DISCUSSION TITLE
Restarting view due to psutil going stale...

### DISCUSSION THREAD

**Discussion by kchang4** [swar/Swar-Chia-Plot-Manager#106] [Q&A]
Happens about 10 hours ish after initial run? Then it'll happen every 20-30mins ish after that.
Problem is that it doesn't actually end up restarting the view, it basically freezes the terminal

Seems like it has to do with this portion of the code

```
// commands.py
has_file = False
  if len(running_work.values()) == 0:
    has_file = True
  for work in running_work.values():
    if not work.log_file:
      continue
    has_file = True
    break
```

I can probably try running a debugging process and see which is setting it false, but do you have a hunch?

I'm on windows 10, running this through the windows terminal.  This happens regardless if I run after activating the venv or not

**Comment by tanvrkhan** [comment]
I have the same issue.

**Comment by garthheff** [comment]
i started getting this since updating to chia 1.1.3

**Comment by swar** [comment]
I think you should restart and try again. This issue is tied to some windows bug that some people have. View is meant to be fallible. Manager however should be resilient to errors. So if view breaks, it should still let manager run in the background.

**Reply by Ron-ski** [reply]
Has there been any progress on finding a fix for this?

I've now got this problem on one of my plotters, the view keeps stopping, with the error "Restarting view due to psutil going stale", when it does restart or if I start it manually then it sees all the jobs, but shows them all as stage 1, zero progress and zero size. View has been rather slow to update for a while (taking well over a couple of minutes), but I only started to get this error today. Googling doesn't have any answers, just a few others having the same issue. This is on Windows 10, my other plotter is also WIndows 10, but that's running fine.

**Reply by Ron-ski** [reply]
Reboot seemed to cure it, at least for now. So may just reboot between each plotting session.

**Reply by christo-m** [reply]
After a day or so 'status/view' will start showing all my plots at stage 1 with no plot_id. At this point no new plots will start until the max plots allowed in phase 1 rule passes. This breaks the queue system as all plots are always phase 1. It is not just the 'view' that is broken, it affects manager.

Bottom line is I have the same issue only on my one plotter. I have been trying to troubleshoot it with no luck. At this point all I can do is let the queue finish and restart.

**Reply by sagat64** [reply]
I have same probrem. 

>This issue is tied to some windows bug that some people have.
How can I fix the bug?

**Comment by jasonmilstead** [comment]
This is what solved this issue for me: Open up Task Manager and go to your Performance tab. Then, under CPU check your Handles. I found that mine were in the millions. That told me that a process on my machine was having a handle leak. 

Did some googling to find the culprit and came up with this program to help me snuff it out:  Microsoft's Sysinternal - Process Explorer. Once I installed Process Explorer, I was able to filter and sort the active processes by handle usage. I found the offending process and killed it. I was then able to restart the plot viewer with no issues. Seems that psutils will crash on having to many handles open. That worked for me but YMMV.

**Reply by Ron-ski** [reply]
Thanks for the info, I rebooted last night, so we'll see how it goes over the weekend. Already have Process Explorer installed, and have turned on handle count (Select columns/Process Performance tab).

**Reply by Ron-ski** [reply]
Just found my other PCwith the error, funny thing was that it was Process Explorer that had over 6 million handles, closed that and the view came back to life, took a couple of refreshes so properly show all plots though. So I think @jasonmilstead you are onto something there. I've been running Process Explorer prior to you mentioning it, so it could well have caused the issue on my other PC, which also has Process Explorer open but has no issues at the moment. Only difference was I was using the filter to see just the Chia processes on the one that had the issue today.

**Comment by ZielReaver** [comment]
Thanks for the information I was having the same issue and used the process explorer to hunt the problem down. For me it was a part of MSI Dragon Center. The mystic lights option inside the program has a .exe called ledkeeper2.exe and that was what was holding all the handles and causing the error on my viewer. Just wanted to add in my finding sense this post did help me fix the issue wanted to help improve it some for others who come looking later.
